src/PostgREST/ApiRequest.hs

```diff
@@ -184,9 +179,8 @@ userApiRequest schema req reqBody
               ["rpc", proc] -> TargetProc
                               $ QualifiedIdentifier schema proc
               other         -> TargetUnknown other
-  shouldParsePayload = action `elem` [ActionCreate, ActionUpdate, ActionInvoke{isReadOnly=False}]
-  relevantPayload | action == ActionInvoke{isReadOnly=True} = Nothing
-                  | shouldParsePayload = rightToMaybe payload
+  shouldParsePayload = action `elem` [ActionCreate, ActionUpdate, ActionInvoke{isReadOnly=False}, ActionInvoke{isReadOnly=True}]
+  relevantPayload | shouldParsePayload = rightToMaybe payload
```

Comment:
Wondering why we still have the notion of `relevantPayload`? AFAICT lazy evaluation would allow us to set `iPayload` directly to `payload`, which would not get evaluated if the app didn't request it.

Reply:
I think if we remove that some of the pattern matching relying on a `Nothing` PayloadJSON in `App.hs` would not work. I think that part would be better refactored in another PR as it's not tied directly to this PR changes, also it already took me a lot of work to refactor the PayloadJSON and I'd rather not touch this part for now.